lib/dayron/config.ex

```diff
@@ -10,14 +10,16 @@ defmodule Dayron.Config do
   def parse(repo, opts) do
     otp_app = Keyword.fetch!(opts, :otp_app)
     config  = Application.get_env(otp_app, repo, [])
-    url = opts[:url] || config[:url]
     adapter = opts[:adapter] || config[:adapter] || Dayron.HTTPoisonAdapter
 
-    unless url do
-      raise ArgumentError, "missing :url configuration in " <>
-                           "config #{inspect otp_app}, #{inspect repo}"
+    case parse_url(opts[:url] || config[:url]) do
+      {:ok, nil} -> raise ArgumentError, "missing :url configuration in " <>
```

Comment:
minor: 
why are you returning `{:ok, nil}`, I think `:no_url` or `:missing_url` would be a bit cleaner